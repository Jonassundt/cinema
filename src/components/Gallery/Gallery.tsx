import React, { useContext } from "react";

import Parameters from "../../Parameters";
import posters from "../Images";

import posterImages from "../Images";

import styles from "./gallery.module.css";

interface indexes {
  [key: string]: any;
}

function Gallery() {
  const { params, setParams } = useContext(Parameters);

  const setPosterIndex = (posterIndex: number) => {
    if (params.loading) return;
    sessionStorage.setItem("posterIndex", posterIndex.toString());
    setParams({ ...params, posterIndex, slideshow: false });
    window.scroll({
      top: 0,
      behavior: 'smooth'
    });
  }

  return (
    <div className={styles.Gallery}>
      {posterImages.filter((poster, index) => params.favorites.includes(index) || !params.showFavorites).map((poster, index) => (
        <div
          className={styles.Thumbnail}
          key={index}
          onClick={() => setPosterIndex(index)}
        >
          <div className={styles.SmallPoster}>{poster}</div>
        </div>
      ))}
    </div>
  );
}

export default Gallery; 
