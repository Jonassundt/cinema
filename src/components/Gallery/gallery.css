.Gallery {
    display: flex;
    margin-bottom: 100px;
}

.Thumbnail {
    height: 100px;
    width: 100px;
    background-color: rgb(0, 84, 87);
    border-radius: 12px;
    padding: 12px;    
    margin-right: 30px;
}

.Thumbnail:hover {
    filter: opacity(70%);
    cursor: pointer;
}