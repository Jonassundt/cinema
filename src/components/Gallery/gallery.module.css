.Gallery {
    display: flex;
    margin-top: 80px;
    margin-bottom: 100px;
}

.Thumbnail {
    height: 400px;
    background: #3e3f47ce;
    border-radius: 20px;
    margin-right: 30px;
    box-shadow: -20px 0 60px #000;
    transition: 0.2s;
}

.Thumbnail:hover {
    cursor: pointer;
    transform: translateY(-20px) skew(-1deg);
    filter: brightness(110%)
}

.Thumbnail:not(:hover) {
    filter: brightness(70%)
}

.Thumbnail:hover~.Thumbnail {
    transform: translateX(120px);
}

.Thumbnail:not(:first-child) {
    margin-left: -120px;
}

.Thumbnail:hover div {
	position: relative;
}

.Thumbnail:hover div::before {
	position: absolute;
	top: 0;
	left: -95%;
	z-index: 2;
	display: block;
	content: '';
	width: 50%;
	height: 100%;
	background: -webkit-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,.3) 100%);
	background: linear-gradient(to right, rgba(255,255,255,0) 0%, rgba(255,255,255,.3) 100%);
	-webkit-transform: skewX(-25deg);
	transform: skewX(-25deg);
}

.Thumbnail:hover div:hover::before {
	-webkit-animation: shine .75s;
	animation: shine .75s;
}

@-webkit-keyframes shine {
	100% {
		left: 125%;
	}
}

@keyframes shine {
	100% {
		left: 125%;
	}
}

.SmallPoster {
    border-radius: 20px;
    overflow: hidden;
}
