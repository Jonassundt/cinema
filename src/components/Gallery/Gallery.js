import React from 'react';

import './gallery.css';

function Thumbnail() {
    return (
        <div className="Thumbnail">
            Artwork
        </div>
    )
}

function Gallery() {
    return (
        <div className="Gallery">
            {[0, 1, 2, 3].map(image => <Thumbnail />)}
            <p>
                TODO: Show all artworks here, for the user to hover and click
            </p>
        </div>
    );
}

export default Gallery;
