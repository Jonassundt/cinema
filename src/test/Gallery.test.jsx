import React from 'react';
import renderer from 'react-test-renderer';
import Gallery from '../components/Gallery/Gallery.tsx';

describe('testing the gallery', () => {
    it("matches the snapshot", () => {
        const tree = renderer.create(<Gallery />).toJSON();

        expect(tree).toMatchSnapshot();
    })
});