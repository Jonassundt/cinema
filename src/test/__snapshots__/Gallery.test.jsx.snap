// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the gallery matches the snapshot 1`] = `
<div
  className="Gallery"
>
  <div
    className="Thumbnail"
    onClick={[Function]}
  >
    <div
      className="SmallPoster"
    >
      <svg
        height={380}
        name="Jamesbond"
        width={285.0712678169542}
      >
        jamesbond.svg
      </svg>
    </div>
  </div>
  <div
    className="Thumbnail"
    onClick={[Function]}
  >
    <div
      className="SmallPoster"
    >
      <svg
        height={380}
        name="Inception"
        width={285.0712678169542}
      >
        inception.svg
      </svg>
    </div>
  </div>
  <div
    className="Thumbnail"
    onClick={[Function]}
  >
    <div
      className="SmallPoster"
    >
      <svg
        height={380}
        name="Interstellar"
        width={285.0712678169542}
      >
        interstellar.svg
      </svg>
    </div>
  </div>
  <div
    className="Thumbnail"
    onClick={[Function]}
  >
    <div
      className="SmallPoster"
    >
      <svg
        height={380}
        name="Batman"
        width={285.0712678169542}
      >
        batman.svg
      </svg>
    </div>
  </div>
  <div
    className="Thumbnail"
    onClick={[Function]}
  >
    <div
      className="SmallPoster"
    >
      <svg
        height={380}
        name="Shawshank"
        width={285.0712678169542}
      >
        shawshank.svg
      </svg>
    </div>
  </div>
  <div
    className="Thumbnail"
    onClick={[Function]}
  >
    <div
      className="SmallPoster"
    >
      <svg
        height={380}
        name="Starwars"
        width={285.0712678169542}
      >
        starwars.svg
      </svg>
    </div>
  </div>
</div>
`;
